includes/User.php

```diff
@@ -4841,7 +4841,29 @@ protected function saveOptions() {
 			}
 		}
 
-		$dbw->delete( 'user_properties', array( 'up_user' => $this->getId() ), __METHOD__ );
+		// kinda ghetto, but :(
+		if ($wgPreferencesUseService) {
+			$regex = implode ('|', $wgUserPreferenceWhiteList['regexes']);
```

Comment:
In which case does this need to handle regular expressions? I think the goal of this piece of functionality is "delete entries in `user_properties` for this user that are not preferences", which can be achieved without the use of the whitelist, since the entries for a user are stored by the preference service. It also doesn't need to know about regular expressions IMO because it will do a straight forward `NOT IN (some-pref-12345, some-pref-23456)` for those properties.